src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -125,6 +138,80 @@ public static object GetClassFactoryForType(ComActivationContext cxt)
             return new BasicClassFactory(cxt.ClassId, classType);
         }
 
+        /// <summary>
+        /// Entry point for unmanaged COM register/unregister API from managed code
+        /// </summary>
+        /// <param name="cxt">Reference to a <see cref="ComActivationContext"/> instance</param>
+        /// <param name="register">true if called for register or false to indicate unregister</param>
+        public static void ClassRegisterationScenarioForType(ComActivationContext cxt, bool register)
+        {
+            // Retrieve the attribute type to use to determine if a function is the requested user defined
+            // registration function.
+            string attributeName = register ? "ComRegisterFunctionAttribute" : "ComUnregisterFunctionAttribute";
+            Type? regFuncAttrType = Type.GetType($"System.Runtime.InteropServices.{attributeName}, System.Runtime.InteropServices", throwOnError: false);
```

Comment:
Given the recent issues with contextual reflection (which this is one of the affected APIs):
* Does this mean that `System.Runtime.InteropServices` assembly is not always available? I find that really weird. I guess the question is, in which case this would return null. I'm wondering why we can't take a static typeref here.
* Can we use either the `Assembly.GetType` or provide an assembly resolver - this ties to what ALC will perform the assembly for the `System.Runtime.InteropServices` (if we actually do have to load it dynamically). Since that is very likely a framework assembly we want the default ALC to load it - always I assume. On the other hand, for best compatibility, loading it through the ALC of the component we're registering would work as well. The objection is that we should be explicit about it.

Reply:
I didn't throw just because I didn't really need to in this case. I am loading the user's type and if the attribute isn't there, then there is no reason to throw because that type clearly isn't adding it to their class. I can throw if that is preferred but in this case it just means they didn't add it to their class. I can be explicit if that is preferred though.

Reply:
If CoreLib depends on this attribute, we should better move it to CoreLib to avoid this whole problem.

Reply:
It doesn't really depend on it. This is from the otherside. Does the type in question use it and if it does it will be found, otherwise it won't and we can move on. I was following the same logic used for the License types below (i.e. `System.ComponentModel.LicenseProviderAttribute` and `System.ComponentModel.LicenseException`).

Reply:
The `System.ComponentModel.License*` has a significant dependency closure. We try to avoid faulting higher up framework parts with lot of dependencies into CoreLib.

`ComRegisterFunctionAttribute` does not depend on anything. It does not have the large dependency closure problem. Faulting it into CoreLib would not be a problem.

(It is ok to use reflection here if you prefer. It does not matter that much.)

Reply:
Thanks for the explanation. I agree that it doesn't matter much, using reflection is perfectly fine here.
If we do, let's be explicit about ALC though.

Reply:
>The System.ComponentModel.License* has a significant dependency closure. We try to avoid faulting higher up framework parts with lot of dependencies into CoreLib.

Okay I missed that nuance. Is there a prior commit that demonstrates how to move a type from a public assembly into SPCL and properly forward it? I am more than willing to do it, but have no idea on the process.

Reply:
>  Is there a prior commit that demonstrates how to move a type from a public assembly into SPCL and properly forward it? 

1. Add the type to CoreLib
2. Delete the type in CoreFX once CoreCLR with that change gets there

Reply:
Boo. This is getting to be a rats nest of required changes and will require this PR to wait for flow from corefx into coreclr and will simply take a lot of time. I have filed https://github.com/dotnet/coreclr/issues/24476 to track the desire to fold some types from assemblies in corefx into SPCL.